fe/fe-core/src/main/java/com/starrocks/connector/RemoteScanRangeLocations.java

```diff
@@ -216,7 +221,10 @@ private void createHudiScanRangeLocations(long partitionId,
         TScanRangeLocation scanRangeLocation = new TScanRangeLocation(new TNetworkAddress("-1", -1));
         scanRangeLocations.addToLocations(scanRangeLocation);
 
-        result.add(scanRangeLocations);
+        if (!result.add(scanRangeLocations)) {
```

Comment:
If there are many scan ranges, the performance is very bad of `HashSet.add`. You could use list and shuffle the order.

Reply:
changed